core/proto.go

```diff
@@ -179,6 +187,10 @@ func UnsignedDataSetToProto(set UnsignedDataSet) (*pbv1.UnsignedDataSet, error)
 
 // UnsignedDataSetFromProto returns the set from a protobuf.
 func UnsignedDataSetFromProto(typ DutyType, set *pbv1.UnsignedDataSet) (UnsignedDataSet, error) {
+	if set == nil {
+		return nil, errors.New("unsigned data set proto cannot be nil")
+	}
+
 	resp := make(UnsignedDataSet)
 	for pubkey, data := range set.Set {
```

Comment:
what about set.Set?

Reply:
for range loop over set.Set will take care of that, so no panic. However, I can add explicit check for that too.

Reply:
Done!
Also added some tests.

Reply:
as, set is a map, in that case, probably better to check for len(set.Set) == 0